AppCenter/AppCenter/Internals/Channel/MSChannelDefault.m

```diff
@@ -311,13 +317,54 @@ - (void)suspend {
 }
 
 - (void)resume {
-  if (self.suspended && self.enabled) {
+  __weak typeof(self) weakSelf = self;
+  dispatch_async(self.logsDispatchQueue, ^{
+    typeof(self) strongSelf = weakSelf;
+    [strongSelf resumeSync];
+  });
+}
+
+- (void)resumeSync {
+  if (!self.sender.suspended && self.enabled) {
     MSLogDebug([MSAppCenter logTag], @"Resume channel for group Id %@.", self.configuration.groupId);
     self.suspended = NO;
+    self.stopFlushingCompletion = kMSEmptyStopFlushingCompletion;
     [self flushQueue];
```

Comment:
This is not related to this PR but I don't think this is not necessary.
BTW, if we remove this one, there is a weird behavior which background task gets invalidated with running task.